service/controller/resource/promxy/create.go

```diff
@@ -0,0 +1,46 @@
+package promxy
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/prometheus-meta-operator/service/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	configMap, err := r.getConfigMap(ctx, obj)
+	if err != nil {
+		return microerror.Mask(err)
+	} else if configMap == nil {
+		return nil // Missing config map, we return immediately
```

Comment:
So the expectation is that we'll just re-try on next reconciliation right? Maybe just log a message?

Reply:
Either there's an error or configmap is not nil. But there should not be a case where there's no error and no configmap.

Reply:
Do you mean this should never happen?

Reply:
Well, I was checking in the getConfigMap method if the configMap is not found and logging there to avoid duplication but I can move that in the main resource code: 	
```
if apierrors.IsNotFound(err) {
	    r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("config map %s does not exists", key.PromxyConfigMapName()))
		return nil, nil
	} else if err != nil {
		return nil, microerror.Mask(err)
	}
```

Reply:
Yes please move this here, I see no reason to hide such simple logic.